apis/kueue/v1alpha1/tas_types.go

```diff
@@ -69,6 +69,7 @@ type TopologySpec struct {
 	// +listType=atomic
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="levels is immutable"
```

Comment:
```suggestion
	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="the levels field is immutable"
```
or "levels are immutable"